pkgs/development/guile-modules/guile-reader/0001-fix-prototypes.diff

```diff
@@ -0,0 +1,26 @@
+diff --git a/src/reader.c b/src/reader.c
```

Comment:
Where does this patch come from?

Reply:
I have tried but the script fails with the '&id=...' part of the url. Anyway I have changed it accordingly.

P.S.: is it needed to download the patch? I particularly prefer the patch locally, in order to edit it if necessary.

Reply:
Yes, the id is necessary for immutable file (it is a commit hash).

Patches other than ones specific to Nix have no business being in Nixpkgs. Fixes should always go to upstream and patch obtained from there. Temporarily, we can use patches from other downstreams but the patch needs to be sent upstream eventually.

Reply:
How can I 'escape' the & sign in the URL?

Reply:
Just quote the URL.

Reply:
Using single quotes, 'like this'? It just complains.

Reply:
No, double quotes like regular string:

https://github.com/NixOS/nixpkgs/blob/9424664a91c42719ac62539fefe0cf5c2f843806/pkgs/desktops/deepin/deepin-terminal/default.nix#L15-L20

Reply:
Now I think it's OK. I have tried the double quotes but it failed; now with the "name = " parameter it works.